packages/cli/src/amplify.ts

```diff
@@ -3,27 +3,41 @@
 import { hideBin } from 'yargs/helpers';
 import { createMainParser } from './main_parser_factory.js';
 import { attachUnhandledExceptionListeners } from './error_handler.js';
+import { extractSubCommands } from './extract_subcommands.js';
 import {
+  AmplifyFault,
   PackageJsonReader,
   UsageDataEmitterFactory,
 } from '@aws-amplify/platform-core';
 import { fileURLToPath } from 'node:url';
+import { printer } from '@aws-amplify/cli-core';
 
 const packageJson = new PackageJsonReader().read(
   fileURLToPath(new URL('../package.json', import.meta.url))
 );
-const libraryVersion = packageJson.version ?? '';
+const libraryVersion = packageJson.version;
+
+if (libraryVersion == undefined) {
+  throw new AmplifyFault('UnknownVersionFault', {
+    message:
+      'Library version cannot be determined. Check the library installation',
+  });
+}
 
 const usageDataEmitter = await new UsageDataEmitterFactory().getInstance(
   libraryVersion
 );
 
 attachUnhandledExceptionListeners(usageDataEmitter);
 
-const parser = createMainParser(usageDataEmitter);
+const parser = createMainParser(libraryVersion, usageDataEmitter);
+await parser.parseAsync(hideBin(process.argv));
 
-await parser
-  .parseAsync(hideBin(process.argv))
-  .then(async (argv) =>
-    usageDataEmitter.emitSuccess({}, { command: argv._.join(' ') })
+try {
+  await usageDataEmitter.emitSuccess(
+    {},
+    { command: extractSubCommands(parser) }
   );
+} catch (e) {
+  printer.log('Failed to emit usage metrics');
```

Comment:
```suggestion
  printer.debug('Failed to emit usage metrics');
```
And also print out what the error was